src/stactools/noaa_hrrr/stac.py

```diff
@@ -0,0 +1,556 @@
+
+GRIB2_MEDIA_TYPE = "application/wmo-GRIB2"
+NDJSON_MEDIA_TYPE = "application/x-ndjson"
+INDEX_ASSET_DEFINITION = AssetDefinition(
+    {
+        "type": NDJSON_MEDIA_TYPE,
+        "roles": ["index"],
+        "title": "Index file",
+        DESCRIPTION: (
+            "The index file contains information on each message within "
+            "the GRIB2 file."
+        ),
+    }
+)
+
+ITEM_BASE_ASSETS = {
+    Product.sfc: {
+        ItemType.GRIB: AssetDefinition(
+            {
+                "type": GRIB2_MEDIA_TYPE,
+                "roles": ["data"],
+                "title": "2D Surface Levels",
+                DESCRIPTION: (
+                    "2D Surface Level forecast data as a grib2 file. Subsets of the "
+                    "data can be loaded using the provided byte range."
+                ),
+            }
+        ),
+        ItemType.INDEX: INDEX_ASSET_DEFINITION,
+    },
+    Product.subh: {
+        ItemType.GRIB: AssetDefinition(
+            {
+                "type": GRIB2_MEDIA_TYPE,
+                "roles": ["data"],
+                "title": "2D Surface Levels - Sub Hourly",
+                DESCRIPTION: (
+                    "2D Surface Level forecast data (sub-hourly, 15 minute intervals) "
+                    "as a grib2 file. Subsets of the data can be loaded using the "
+                    "provided byte range."
+                ),
+            }
+        ),
+        ItemType.INDEX: INDEX_ASSET_DEFINITION,
+    },
+    Product.prs: {
+        ItemType.GRIB: AssetDefinition(
+            {
+                "type": GRIB2_MEDIA_TYPE,
+                "roles": ["data"],
+                "title": "3D Pressure Levels",
+                DESCRIPTION: (
+                    "3D Pressure Level forecast data as a grib2 file. Subsets of the "
+                    "data can be loaded using the provided byte range."
+                ),
+            }
+        ),
+        ItemType.INDEX: INDEX_ASSET_DEFINITION,
+    },
+    Product.nat: {
+        ItemType.GRIB: AssetDefinition(
+            {
+                "type": GRIB2_MEDIA_TYPE,
+                "roles": ["data"],
+                "title": "Native Levels",
+                DESCRIPTION: (
+                    "Native Level forecast data as a grib2 file. Subsets of the data "
+                    "can be loaded using the provided byte range."
+                ),
+            }
+        ),
+        ItemType.INDEX: INDEX_ASSET_DEFINITION,
+    },
+}
+
+RENDER_PARAMS = {
+    Product.sfc: {
+        "WIND__10_m_above_ground__periodic_max": {
+            "title": "Wind speed (m/s) 10 m above ground",
+            "colormap_name": "viridis",
+            "rescale": [[0, 20]],
+            "resampling": "nearest",
+        },
+        "DSWRF__surface__point_in_time": {
+            "title": "Downward Short-Wave Radiation Flux (W/m2)",
+            "colormap_name": "rainbow",
+            "rescale": [[0, 800]],
+            "resampling": "nearest",
+        },
+    }
+}
```

Comment:
@abarciauskas-bgse @vincentsarago the render parameters are not complete because we will need to specify the layer-specific `dsn` rather than specify a STAC asset.